### DISCUSSION TITLE
How to get the geometry proprieties (such as pore/throat diameter) of designated pores of throats?

### DISCUSSION THREAD

**Discussion by mourendong** [PMEAL/OpenPNM#1976] [Q&A] [answered]
How to get the geometry proprieties (such as pore/throat diameter) of designated pores of throats?
Firstly, I built a pore network (**pn**) and labeled it with two parts;
Then, the geometry was added to the two parts, geo1 and geo2. 
I can get the pore index from the **pn** part,
for example, the throat index [ 3 24 40],
I want get the geometry proprieties of one pore or throat through the **geo1['pore.diameter'][i]**, i is the index of pore or throat.
But in the geo1 or geo2 part, the index in the range of 20, the order is changed.
How can I get the geometry proprieties? Can you give me some advice? Thanks for your help!

**Comment by ma-sadeghi** [comment, marked as answer]
If I understood your question properly, here's an example:

- `net` consists of pores `[0, 1, 2, 3, 4]`
- `geo1` consists of pores `[0, 2, 3]`
- `geo2` consists of pores `[1, 4]`

If you call `geo1.pores()`, you'll get: `[0, 1, 2]`. Now, if you want to see which indices do these pores correspond to in the network, you can use: `geo1.pores(target=net)`, which will return `[0, 2, 3]`.

More generally, we have `map_pores` method that maps local to global indices and vice versa. Here's how the API looks like:

```python
# Map pores with "network" indices = [0] to local indices in geo1
geo1.map_pores(pores=[0], origin=net)  # should return 0
# Map pores with "network" indices = [1] to local indices in geo1
geo1.map_pores(pores=[0], origin=net)  # should return [] since pore 0 doesn't belong to geo1
# Map pores with "network" indices = [2] to local indices in geo1
geo1.map_pores(pores=[0], origin=net)  # should return [1]
# Map pores with "geo2" indices = [1] to global indices in network
net.map_pores(pores=[1], origin=geo2)  # should return [4] since pore 1 in geo2 is pore 4 in the network
```
Hope this clarifies the issue.

**Reply by mourendong** [reply]
Thanks for your kind reply! As you mentioned,

net consists of pores [0, 1, 2, 3, 4]
geo1 consists of pores [0, 2, 3]
geo2 consists of pores [1, 4]
In geo1, how to get the pore diameter? The diameter in geo1 may index as [0, 1, 2], and in geo2 which is [0,1].

**Reply by mourendong** [reply]
Thanks for your kind help! 
I understand! The question has been solved.
